blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.fetchTweet = this.fetchTweet.bind( this );
+			this.state = {
+				url: this.props.attributes.url,
+				html: '',
+				error: false,
+				fetching: false,
+			};
+		}
+		doFetch( url, setAttributes, setState ) {
+			setState( { fetching: true, error: false } );
+			jQuery.ajax( {
```

Comment:
Hmm, this _could_ be related to #691. Generally I tend to discourage network requests in the component itself. The line is blurred here because the result of this request is not really valuable outside the context of the component.

The use of `jQuery` also sparked my interest. It's handy and readily accessible. We'll probably want to think on our options here some more; probably outside the scope of this pull request specifically.

At the very least, with mind toward #742 and eliminating globals*, we'll probably want to import this as a module and configure Webpack to treat this as an external ([reference](https://github.com/WordPress/gutenberg/blob/1c1e7eab535dcb2581321d291e449f69e2153825/webpack.config.js#L22-L26)). I believe it should be enough to add `'jquery': 'jQuery'` there, and then you can `import jQuery from 'jquery';` at the top of this file.

_*_ For consistency and to enable changing the source from which jQuery is resolved.